### DISCUSSION TITLE
SSH key dropping

### DISCUSSION THREAD

**Discussion by vx-sec** [kovidgoyal/kitty#6891] [Q&A]
Is there a way i could make kitty append an ssh public key to the authorized_keys file of any host i connect to with the trully convenient ssh kitten? One way i thought of is using the COPY to just copy the pub file to the authorized_keys, but that would replace all existing keys on the server.

**Comment by kovidgoyal** [comment]
I dont understand your use case. What's the point of appending keys to
authorized_keys *after* you have already connected with ssh? And if you
want to do so, how will you prevent it from appending the same key
repeatedly. 

But just for completeness, you can run arbitrary code using the ssh
kitten, by copying over a shell rc file and putting whatever code you
want to execute in that file.

**Reply by vx-sec** [reply]
The use case is basically dropping the key on first connect to the server so that subsequent connections will not require a password.
